src/opm/output/eclipse/WriteRFT.cpp

```diff
@@ -0,0 +1,391 @@
+            welletc[etcIx::Viscosity] = "  INPUT";
+            welletc[etcIx::ConcPlyBr] = "  INPUT";
+            welletc[etcIx::PlyBrRate] = "  INPUT";
+            welletc[etcIx::PlyBrAds]  = "  INPUT";
+        }
+    } // namespace RftUnits
+
+    class WellRFT
+    {
+    public:
+        explicit WellRFT(const std::size_t nconn = 0);
+
+        void add(const ::Opm::UnitSystem&       usys,
+                 const ::Opm::Connection&       conn,
+                 const ::Opm::data::Connection& xcon,
+                 const double                   depth);
+
+        std::size_t nConn() const { return this->i_.size(); }
+
+        const std::vector<int>& conI() const { return this->i_; }
+        const std::vector<int>& conJ() const { return this->j_; }
+        const std::vector<int>& conK() const { return this->k_; }
+
+        const std::vector<float>& depth()    const { return this->depth_; }
+        const std::vector<float>& pressure() const { return this->press_; }
+        const std::vector<float>& swat()     const { return this->swat_; }
+        const std::vector<float>& sgas()     const { return this->sgas_; }
+
+        const std::vector<Opm::EclIO::PaddedOutputString<8>>&
+        hostgrid() const
+        {
+            return this->host_;
+        }
+
+    private:
+        std::vector<int> i_;
+        std::vector<int> j_;
+        std::vector<int> k_;
+
+        std::vector<float> depth_;
+        std::vector<float> press_;
+        std::vector<float> swat_;
+        std::vector<float> sgas_;
+
+        std::vector<Opm::EclIO::PaddedOutputString<8>> host_;
+    };
+
+    WellRFT::WellRFT(const std::size_t nconn)
+    {
+        if (nconn == 0) { return; }
+
+        this->i_.reserve(nconn);
+        this->j_.reserve(nconn);
+        this->k_.reserve(nconn);
+
+        this->depth_.reserve(nconn);
+        this->press_.reserve(nconn);
+        this->swat_ .reserve(nconn);
+        this->sgas_ .reserve(nconn);
+
+        this->host_.reserve(nconn);
+    }
+
+    void WellRFT::add(const ::Opm::UnitSystem&       usys,
+                      const ::Opm::Connection&       conn,
+                      const ::Opm::data::Connection& xcon,
+                      const double                   depth)
+    {
+        this->i_.push_back(conn.getI() + 1);
+        this->j_.push_back(conn.getJ() + 1);
+        this->k_.push_back(conn.getK() + 1);
+
+        using M = ::Opm::UnitSystem::measure;
+        auto cvrt = [&usys](const M meas, const double x) -> float
+        {
+            return usys.from_si(meas, x);
+        };
+
+        this->depth_.push_back(cvrt(M::length  , depth));
+        this->press_.push_back(cvrt(M::pressure, xcon.cell_pressure));
+
+        this->swat_.push_back(xcon.cell_saturation_water);
+        this->sgas_.push_back(xcon.cell_saturation_gas);
+
+        this->host_.emplace_back();
+    }
+
+    WellRFT
+    createWellRFT(const int                                 reportStep,
```

Comment:
I think the use of constructions functions like this is unfortunate[1] - that way you impose a value/pointer (in this case value) semantic on the calling scope - which calling scope should be free to determine; i.e. if I want to put the `WellRFT` object in managed pointer I am out of luck here. Probably not very relevant in this particular case - but still.


[1]: Yes - I have written lot's of such functions myself - and changed my ways.

Reply:
I'm hesitant to put the work of `create()` into the constructor because it is essentially unbounded, but I can switch to returning a `unique_ptr<WellRFT>` from the factory function instead.

Reply:
Just to follow up, you could of course also write
```C++
auto rft = createWellRFT(/* ... */);
auto p = std::make_unique<WellRFT>(std::move(rft));
```
which, although inelegant, does at least allow you to put the object into a smart pointer.